src/test/compile-fail/catch-bad-lifetime.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(catch_expr)]
```

Comment:
can you add a comment on this test explaining what the test aims to demonstrate (e.g., that a borrowed valid inside the `do catch` will propagate out?

I also think it'd be good to possibly break this test into distinct files.